jax/_src/checkify.py

```diff
@@ -457,14 +459,23 @@ def assert_impl(pred, code, payload, *, msgs):
   Error(~pred, code, msgs, payload).throw()
   return []
 
-@assert_p.def_abstract_eval
+CheckEffect = enum.Enum('CheckEffect', ['CHECK'])
```

Comment:
do you see this as the preferred type for an effect? I wasn't sure what type to give it (in this case it will be an enum with only one value)

Reply:
It doesn't necessarily have to be an enum -- the main requirement is that it has to be hashable. If you only have one effect (unlike debug printing, which uses two), you can just use a sentinel object.